### ISSUE TITLE
Have a function control the wiki link completion style

### ISSUE THREAD

**Issue by milch** [epwalsh/obsidian.nvim#377]
### 🚀 The feature, motivation and pitch

Currently the way you control the wiki-link completion options is through the 3 mutually exclusive bool options:

```lua
    -- Control how wiki links are completed with these (mutually exclusive) options:
    --
    -- 1. Whether to add the note ID during completion.
    -- E.g. "[[Foo" completes to "[[foo|Foo]]" assuming "foo" is the ID of the note.
    -- Mutually exclusive with 'prepend_note_path' and 'use_path_only'.
    prepend_note_id = true,
    -- 2. Whether to add the note path during completion.
    -- E.g. "[[Foo" completes to "[[notes/foo|Foo]]" assuming "notes/foo.md" is the path of the note.
    -- Mutually exclusive with 'prepend_note_id' and 'use_path_only'.
    prepend_note_path = false,
    -- 3. Whether to only use paths during completion.
    -- E.g. "[[Foo" completes to "[[notes/foo]]" assuming "notes/foo.md" is the path of the note.
    -- Mutually exclusive with 'prepend_note_id' and 'prepend_note_path'.
    use_path_only = false,
```

Personally none of these three options really work well for me, but I think rather than request a fourth option here I it would make sense to just have an option like `wiki_link_complete_func` that returns the string between the double brackets. All three existing options can be accomplished with this assuming the function is passed at least the note id, title and path.  

### Alternatives

- The alternative would be to keep adding more boolean options as users request more ways to construct these 
- Another one would be to support some sort of template string, e.g. `{{path}}|{{id}}`, but that again limits the flexibility to whatever is implemented in this plugin, e.g. if `{{path}}`, `{{title}}` and `{{id}}` are supported, but someone wants a title that looks like `{{top-level-folder}}: {{title}}` then support for `{{top-level-folder}}` needs to be implemented here - you're back on a treadmill to implement every option every user wants

**Comment by epwalsh** [comment]
Hey @milch, this would be great to have. The boolean options are pretty terrible TBH 😭... an unfortunate artifact of how this project evolved. But I think we could implement this in a backwards-compatible way, where each of the boolean options map to a built-in like-generating function.

**Comment by epwalsh** [comment]
Actually this is probably pretty straight-forward to implement now that all link text is generated through a single client method (`Client:format_link(...)`).

**Comment by epwalsh** [comment]
Implemented in #406.

**Comment by milch** [comment]
Nice, thank you!
